dashboard/test/controllers/omniauth_callbacks_controller_test.rb

```diff
@@ -666,6 +666,44 @@ class OmniauthCallbacksControllerTest < ActionController::TestCase
     assert_equal User.last.id, signed_in_user_id
   end
 
+  test 'sign_up_clever: email taken redirect if _two_ users are already using your email address' do
```

Comment:
That second case can't be eliminated for Clever because `User.from_omniauth` _may_ persist a user whose email is already taken, so I wanted to work out test cases that actually exercise this path.

It turns out this only happens if
1. You are signing in with an untrusted email provider, like Clever.
2. You are a teacher so we actually keep and care about your email address.
3. A prior account already exists in our system using your email address.
4. A _second_ prior account also exists that was created with an untrusted email provider (like Clever) resulting in an email like 'me@example.com.oauthemailalreadytaken'.
5. Neither of the above accounts match your OAuth credentials.

For example, this might be an issue if many Clever teacher accounts have the same email address (apparently possible in their system).

For extra weirdness, the resulting redirect and the message we eventually display to the teacher varies depending on whether the second account (created in step 4, above) is a Clever account or some other untrusted email account - maybe a mistake?

I'd eventually like creation to fail at the second account (step 4) above and direct the teacher to perform a password reset on their existing account, and link Clever via the accounts page.  For now though, I'll settle for test coverage of these weird cases.